src/main/java/org/currencygoldexchangeapp/Main.java

```diff
@@ -47,32 +41,73 @@ public static void main(String[] args) {
         }
     }
 
+    private static LocalDate getDate(Scanner scanner) {
+        LocalDate date = null;
+
+        while (date == null) {
+            System.out.print("Enter date (optional, press ENTER for today's date): ");
+            String dateString = scanner.nextLine();
+
+            try {
+                date = dateString.isEmpty() ? LocalDate.now() : LocalDate.parse(dateString);
+            } catch (DateTimeParseException e) {
+                System.out.println("Invalid date format. Please enter a valid date in the format yyyy-MM-dd");
+            }
+        }
+        return date;
+    }
+
+    public static boolean isCurrencyAvailable(String currency, String date) {
+        List<String> codeCurrencyAvailableFor2011AndLater = Arrays.asList("USD", "AUD", "CAD", "EUR", "HUF", "CHF", "GBP", "JPY", "CZK", "DKK", "NOK", "SEK", "XDR");
+        List<String> codeCurrencyAvailableFor2010AndBefore = Arrays.asList("USD", "AUD", "CAD", "EUR", "HUF", "CHF", "GBP", "JPY", "CZK", "DKK", "EEK", "NOK", "SEK", "XDR");
+
+        int year = Integer.parseInt(date.substring(0, 4));
+
+        List<String> availableCurrencies;
+
+        if (year >= 2011) {
+            availableCurrencies = codeCurrencyAvailableFor2011AndLater;
+        } else if (year >= 2003) {
+            availableCurrencies = codeCurrencyAvailableFor2010AndBefore;
+        } else {
```

Comment:
Should you even accept dates outside of this range above? Please include this into the menu validation in future commit

Reply:
Yes I agree, It shouldn't be accepted. Added logic to handle these cases